### PR TITLE
Add WebKitGTK base app

### PR DESCRIPTION
**Pull request by gasinvein**
This is mostly needed for Lutris #926, but may be useful for other apps that need WebKitGTK, but for some reason don't use Gnome runtime, too.

The problem to solve: WebKit is huge and takes a decent amount of time to build. This unnecessarily holds up buildbot queue for a time, and makes local builds too slow, thus complicating maintaining the app.

### CODE REVIEW

**Comment by barthalion** [comment]
Have you asked GNOME/FDO SDK teams if that's something they would be interested doing themselves?

**Comment by gasinvein** [comment]
@barthalion No. I didn't do much on my own here, actually - the build options and the patch are copied from Gnome Sdk.

File: org.webkitgtk.BaseApp.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          ldflags: -Wl,--no-keep-memory
+          config-opts:
+            - -DCMAKE_SYSTEM_PROCESSOR=i586
+    config-opts:
+      - -DPORT=GTK
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DENABLE_PLUGIN_PROCESS_GTK2=OFF
+      - -DENABLE_GTKDOC=OFF
+      - -DENABLE_MINIBROWSER=OFF
+      - -DUSE_WOFF2=OFF
```
Comment (mcatanzaro) [inline org.webkitgtk.BaseApp.yml]: Disabling woff2 font support is really not OK.

**Comment by TingPing** [comment]
WebKitGTK is security sensitive and being in a baseapp means apps will be very delayed in getting updates in general and thats assuming we have a good maintainer of the baseapp in the first place.

**Comment by mcatanzaro** [comment]
I'm surprised you're bundling so many security-critical dependencies (openjdk, libsecret)... not to mention WebKitGTK itself. You're going to keep all these tarballs regularly updated?

**Comment by gasinvein** [comment]
@TingPing I just wanted to split WebKitGTK out of Lutris bundle, and surely I'm not a good maintainer for WebKitGTK itself, especially if it's really going to be reused by other apps.

**Comment by TingPing** [comment]
Maybe Lutris should just depend on the GNOME runtime.

**Comment by gasinvein** [comment]
@TingPing we've started with it, but it has proven to be problematic. Lutris largely depends on multiarch, but the FD.o compat32 extension is not always compatible with Gnome runtime, and already broke once https://github.com/flathub/flathub/pull/926#issuecomment-478542455.

**Comment by TingPing** [comment]
> but the FD.o compat32 extension is not always compatible with Gnome runtime

Well *obviously* the fdo bits shouldn't be used. You need a GNOME runtime equivalent which means having a conversation with them: https://gitlab.gnome.org/GNOME/gnome-build-meta

**Comment by gasinvein** [comment]
Well, will it be acceptable if I add a readme with disclaimer, clearly stating that this baseapp is intended for Lutris flatpak, may be outdated and insecure and, thus, shouldn't be used in other apps?

**Comment by bilelmoussaoui** [comment]
> Well, will it be acceptable if I add a readme with disclaimer, clearly stating that this baseapp is intended for Lutris flatpak, may be outdated and insecure and, thus, shouldn't be used in other apps?

Well, if it's going to.be used by Lutris Flatpak only, there are no needs to have it as a separate baseApp. You can just bundle those dependencies on Lutris's manifest.

**Comment by nedrichards** [comment]
@gasinvein I don't see how that addresses the downsides mentioned above for the users of Lutris. Especially when you get cloud build infra for PRs etc when added to flathub.

**Comment by gasinvein** [comment]
@nedrichards I believe these downsides don't affect Lutris users, at least currently. Lutris uses webkit only for GOG.com login page, it's can't open arbitary webpages.

**Comment by gasinvein** [comment]
bot, build org.webkitgtk.BaseApp

**Comment by mcatanzaro** [comment]
brotli, woff2, and openjpeg are all seriously security-sensitive as well. If you add this app, you're going to have to be *really* on top of it.

**Comment by gasinvein** [comment]
@mcatanzaro It there any way to automate this? Maybe some bot could watch for upstream releases and open open an issue as there are new releases?

**Comment by mcatanzaro** [comment]
Well https://release-monitoring.org/ exists to flag new releases, but you'd need to set up your own infrastructure to hook up to it.

**Comment by mcatanzaro** [comment]
Also some of these modules have CVEs fixed without releases (e.g. openjpeg). Bundling creates a lot to keep track of....

**Comment by TingPing** [comment]
> Well, will it be acceptable if I add a readme with disclaimer, clearly stating that this baseapp is intended for Lutris flatpak, may be outdated and insecure and, thus, shouldn't be used in other apps?

This logic is very backwards. The point of baseapps is to share.

**Comment by gasinvein** [comment]
@TingPing @mcatanzaro Opened [an issue](https://gitlab.gnome.org/GNOME/gnome-build-meta/issues/168) asking Gnome team to provide `org.gnome.Platform.Compat.i386`.

**Comment by gasinvein** [comment]
bot, build org.webkitgtk.BaseApp

File: org.webkitgtk.BaseApp.yml [hunk 2]
```diff
@@ -25,72 +25,85 @@ modules:
...
+      cxxflags: -g1
     config-opts:
       - -DPORT=GTK
       - -DCMAKE_BUILD_TYPE=Release
       - -DENABLE_PLUGIN_PROCESS_GTK2=OFF
       - -DENABLE_GTKDOC=OFF
       - -DENABLE_MINIBROWSER=OFF
+      - -DENABLE_GAMEPAD=OFF
       - -DPYTHON_EXECUTABLE=/usr/bin/python3
+      - -DUSE_WPE_RENDERER=OFF
```
Comment (mcatanzaro) [inline org.webkitgtk.BaseApp.yml:38]: This is important...

Comment (gasinvein) [reply org.webkitgtk.BaseApp.yml]: Thanks, added libwpe.

**Comment by gasinvein** [comment]
bot, build org.webkitgtk.BaseApp

**Comment by gasinvein** [comment]
Telegram 2.7.2, which depends in WebKitGTK, is out, we'll need this soon.

**Pull request closed without merging**